src/preamble.js

```diff
@@ -1017,9 +1019,9 @@ function enlargeMemory() {
 
   var LIMIT = Math.pow(2, 31); // 2GB is a practical maximum, as we use signed ints as pointers
                                // and JS engines seem unhappy to give us 2GB arrays currently
-  if (DYNAMICTOP >= LIMIT) return false;
+  if (HEAPU32[DYNAMICTOP_PTR>>2] >= LIMIT) return false;
```

Comment:
but below we do want `<=` as we want to keep enlarging to a higher value than is needed, to try to avoid extra growth events